ecosystem/sep-0008.md

```diff
@@ -248,14 +248,20 @@ Name | Type | Description
 }
 ```
 
-###### Action URL
-If the `action_method` is provided, the client uses the type of request indicated when sending a request to the `action_url`. If `action_fields` are to be provided and the fields contain sensitive or personal information, the `POST` case is preferred to ensure sensitive information is not transmitted in the URL.
+###### Following the Action URL
+If the `action_method` field is provided, the client uses the method specified when making the request to the `action_url`.
 
-- In the case that `action_method` is `GET`, any fields requested may be passed as query parameters in the `action_url`.
+If the `action_fields` field is provided, the client may optionally supply the fields if they already have the information so as to reduce the user's need to re-enter the information. If the requested fields contain sensitive or personal information, the server should use `action_method` `POST` so that sensitive information is not transmitted in a URL.
 
-- In the case that `action_method` is `POST`, any fields requested may be passed as JSON in the request body, using content type `application/json`. The server should respond to the `POST` with a HTTP response with status code `200`, content-type `application/json`. The body should contain either an acknowledgement that the `POST` was sufficient and no further action is required, or the next URL that the user should be taken to if action is still required.
+There are two possible values for `action_method`.
 
-   The next URL in the POST case is necessary to support products that are websites and products that are mobile applications. Products that run entirely in the browser can redirect the user to the next URL and the redirect prevents odd user experiences where a user refreshes a page and is prompted to re-post the data. Products that are mobile applications cannot open a system browser with a POST request. They can make the POST request internally, extract the redirect in the response, and open a system browser with the next URL. Separating the next URL also makes it possible for the server to indicate that the fields provided in the POST request were sufficient and no further user interaction is required.
+- In the case that `action_method` is `GET`, or not defined:
+
+   Fields requested may be passed as query parameters in the `action_url`. The URL should be opened in a browser. Any fields passed should allow the server to proceed without collecting user information, or at the least pre-fill the provided information.
```

Comment:
> The URL should be opened in a browser.

Is it in the context of web applications? In the case of mobile application, a webview might be more relevant? Perhaps we shouldn't dictate where (browser or mobile apps) and how (popups/webview or another page) the URL is opened?

Reply:
A webview is a type of browser, an embeddable browser. I agree SEP-8 shouldn't dictate UX, it should defer that to implementers as much as possible which is why it doesn't discuss popups, webviews, etc.

Reply:
Any tips for alternative words to use than browser are welcome, let me know if you have a better term.

Reply:
I was thinking we can remove that particular sentence. Is the purpose of including it to differentiate the case of POST, which can happen behind the scene? For non-sensitive information or GET requests in general, can't it happen behind the scene?

Reply:
I added it because you expressed some confusion on our call about which request was displayed to the user and which wasn't, so I added the statements about which are opened in a browser to try and address that. I think it adds clarity.

>For non-sensitive information or GET requests in general, can't it happen behind the scene?

No, the action_required response is to indicate some user action is required. That could be to provide information, or to agree/sign an agreement. The POST case just adds the ability to potentially circumvent that early.

If I'm missing what you mean could you open a PR against this PR to demonstrate?

Reply:
Feel free to merge it.

Just a final confirmation:
Is it correct that we expect the clients to always GET in the front in a browser and POST in the background? I understand the interactive flow with GET is the original way of doing things. It's just that with the addition of POST, which is expected to happen in the background, it wasn't super clear to me whether clients can use GET in the background and use POST in the front (browser). Readers would have to understand that purpose of the addition of POST is to circumvent the interactive flow, not just to send sensitive information.

Reply:
What you said is correct. There is no change to the use case of GET in the foreground. The doc mentions it exists to circumvent the interactive flow, but let me know if we can say more to make that more obvious.